app/components/createSecret/createSecretController.js

```diff
@@ -0,0 +1,67 @@
+angular.module('createSecret', [])
+.controller('CreateSecretController', ['$scope', '$state', 'Messages', 'Secret',
+function ($scope, $state, Messages, Secret) {
+  $scope.formValues = {
+    Name: '',
+    Data: '',
+    Labels: []
+  };
+  $scope.encodeSecret = true;
```

Comment:
`encodeSecret` should be added in the `$scope.formValues` object. Otherwise, when binding it using `ng-model` on the checkbox, it will not be updated.

See: http://stackoverflow.com/a/23943930/1127536